### PR TITLE
[SPARK-19064][PySpark]Fix pip installing of sub components

### PR DESCRIPTION
**Pull request by holdenk**
## What changes were proposed in this pull request?

Fix instalation of mllib and ml sub components, and more eagerly cleanup cache files during test script & make-distribution.

## How was this patch tested?

Updated sanity test script to import mllib and ml sub-components.

### CODE REVIEW

**Comment by holdenk** [comment]
cc @JoshRosen who reviewed the original PR and is probably the most familiar.

File: dev/pip-sanity-check.py [hunk 1]
```diff
@@ -18,6 +18,8 @@
 from __future__ import print_function
 
 from pyspark.sql import SparkSession
+from pyspark.ml.param import Params
+from pyspark.mllib.linalg import *
```
Comment (viirya) [inline dev/pip-sanity-check.py]: Is it better to import `pyspark.ml.linalg` or both?

Comment (holdenk) [inline dev/pip-sanity-check.py]: This just checks one sub component from each, we could import each with rename I suppose but not sure it would do much?

**Comment by holdenk** [comment]
Gentle ping for @joshrosen to @davies maybe?

**Comment by holdenk** [comment]
Small follow up ping for @joshrosen to @davies maybe?

**Review by srowen — APPROVED** [review]
I'd say you can merge it yourself !

**Comment by felixcheung** [comment]
Holden, https://github.com/apache/spark/blob/master/dev/merge_spark_pr.py works well for merging to master and back port to any branch :) unless there is conflict then it would be easier with a separate PR.

Have fun!

File: dev/pip-sanity-check.py (hunk 1 above)
Comment (viirya) [inline dev/pip-sanity-check.py]: ok. i think this should be enough.

**Comment by holdenk** [comment]
Merged into master & branch-2.1

**Pull request closed without merging**